File: app/src/main/java/com/k4dima/chat/core/api/MockingInterceptor.java
package com.k4dima.chat.core.api;

import android.os.SystemClock;
import android.support.annotation.NonNull;

import com.k4dima.chat.core.di.messages.MessagesScope;

import java.security.SecureRandom;
import java.util.Random;

import javax.inject.Inject;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

@MessagesScope
public class MockingInterceptor implements Interceptor {
    private final RequestsHandler handler;
    private final Random random = new SecureRandom();

    @Inject
    MockingInterceptor(RequestsHandler handler) {
        this.handler = handler;
    }

    @Override
    public Response intercept(@NonNull Chain chain) {
        Request request = chain.request();
        Response response = handler.proceed(request);
        int stubDelay = 500 + random.nextInt(1500);
        SystemClock.sleep(stubDelay);
        return response;
    }
}

File: app/src/main/java/com/k4dima/chat/core/api/MockingInterceptor.kt
package com.k4dima.chat.core.api

import android.os.SystemClock
import com.k4dima.chat.features.messages.di.MessagesScope
import okhttp3.Interceptor
import okhttp3.Response
import java.security.SecureRandom
import javax.inject.Inject

@MessagesScope
class MockingInterceptor @Inject
internal constructor(private val handler: RequestsHandler) : Interceptor {
    private val random = SecureRandom()

    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val response = handler.proceed(request)
        val stubDelay = 500 + random.nextInt(1000)
        SystemClock.sleep(stubDelay.toLong())
        return response
    }
}